### PR TITLE
CANDLEPIN-800: Create entitlement certificate generator

### PR DESCRIPTION
**Pull request by Januson**
- Added Entitlement certificate generator
- Added Entitlement payload generator

### CODE REVIEW

File: src/main/java/org/candlepin/pki/certs/EntitlementPayloadGenerator.java [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+        }
+
+        String addons = product.getAttributeValue(Product.Attributes.ADDONS);
+        if (StringUtils.isNotBlank(addons)) {
+            toReturn.setAddons(Util.toList(addons));
+        }
+        return toReturn;
+    }
+
+    private Service createService(Pool pool) {
```
[the remark this answers is not in the document]
Comment (Januson) [reply src/main/java/org/candlepin/pki/certs/EntitlementPayloadGenerator.java:150]: Good catch. Those are used in one test I missed. I'll move it as well and remove them.

File: src/main/java/org/candlepin/pki/certs/EntitlementCertificateGenerator.java [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+        this.consumerTypeCurator = Objects.requireNonNull(consumerTypeCurator);
+        this.environmentCurator = Objects.requireNonNull(environmentCurator);
+        this.keyPairGenerator = Objects.requireNonNull(keyPairGenerator);
+        this.pemEncoder = Objects.requireNonNull(pemEncoder);
+        this.signer = Objects.requireNonNull(signer);
+        this.certificateBuilder = Objects.requireNonNull(certificateBuilder);
+    }
+
+    /**
+     * @param consumer
```
[the remark this answers is not in the document]
Comment (Januson) [reply src/main/java/org/candlepin/pki/certs/EntitlementCertificateGenerator.java:140]: Done.

File: src/main/java/org/candlepin/guice/CandlepinModule.java [hunk 3]
```diff
@@ -309,9 +307,10 @@ private void bindPki() {
         bind(X509CertificateBuilder.class).toProvider(X509CertificateBuilderProvider.class);
 
         bind(AnonymousCertificateGenerator.class);
-        bind(SCACertificateGenerator.class);
+        bind(EntitlementCertificateGenerator.class);
```
[the remark this answers is not in the document]
Comment (Januson) [reply src/main/java/org/candlepin/guice/CandlepinModule.java:311]: Added. It looks like it's not needed for simple bindings, but at least we are explicit about which classes we want to bind.

File: src/main/java/org/candlepin/controller/PoolManager.java [hunk 4]
```diff
@@ -133,7 +133,7 @@ public PoolManager(
         EntitlementCurator entitlementCurator,
         ConsumerCurator consumerCurator,
         ConsumerTypeCurator consumerTypeCurator,
-        EntitlementCertificateGenerator ecGenerator,
+        EntitlementCertificateService ecGenerator,
```
Comment (joshmalbrecht) [inline src/main/java/org/candlepin/controller/PoolManager.java:136]: We can rename this from `ecGenerator` now that the class name has changed

This also applies to:
- ProductManager.java
- ConsumerResource.java
- EnvironmentResource.java
- RegenEnvEntitlementCertsJobTest.java
- RegenProductEntitlementCertsJobTest.java
- EntitlementCertificateServiceTest.java
- PoolManagerFunctionalTest.java
- PoolManagerTest.java
- ProductManagerTest.java
- AnonymousConsumerResourceCreationTest.java
- ConsumerContentOverrideResourceTest.java
- ConsumerResourceCreationTest.java
- ConsumerResourceTest.java
- ConsumerResourceUpdateTest.java
- EnvironmentResourceTest.java

File: src/main/java/org/candlepin/pki/certs/EntitlementPayloadGenerator.java [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+        return toReturn;
+    }
+
+    private String toJson(Object anObject) {
+        String output = "";
+        try {
+            output = this.mapper.writeValueAsString(anObject);
+        }
+        catch (Exception e) {
+            log.error("Could no serialize the object to json " + anObject, e);
```
Comment (joshmalbrecht) [inline src/main/java/org/candlepin/pki/certs/EntitlementPayloadGenerator.java:194]: Typo: `Could no` to `Could not`

**Review by joshmalbrecht — CHANGES_REQUESTED** [review]
Noticed some small changes

File: src/main/java/org/candlepin/pki/certs/EntitlementPayloadGenerator.java (hunk 5 above)
Comment (Januson) [reply src/main/java/org/candlepin/pki/certs/EntitlementPayloadGenerator.java]: Done.

File: src/main/java/org/candlepin/controller/PoolManager.java (hunk 4 above)
Comment (Januson) [reply src/main/java/org/candlepin/controller/PoolManager.java]: Done.

File: src/main/java/org/candlepin/pki/certs/EntitlementPayloadGenerator.java [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+        }
+
+        if (StringUtils.isNotBlank(pool.getAccountNumber())) {
+            toReturn.setAccount(pool.getAccountNumber());
+        }
+
+        return toReturn;
+    }
+
+    private org.candlepin.model.dto.Pool createPool(Pool pool) {
```
Comment (nikosmoum) [inline src/main/java/org/candlepin/pki/certs/EntitlementPayloadGenerator.java:182]: there's a duplicate of this in X509V3ExtensionUtil.java still; lets remove it (I don't think its being used there)

**Review by nikosmoum — APPROVED** [review]

**Review by joshmalbrecht — APPROVED** [review]

**Pull request merged**